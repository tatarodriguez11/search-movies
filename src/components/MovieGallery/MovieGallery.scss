.gallerySlide{
  .gallerySlide-image {
      width: 100%;
  }
}