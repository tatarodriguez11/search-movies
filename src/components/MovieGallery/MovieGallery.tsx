import Slider from 'react-slick';
import './MovieGallery.scss'


const MovieGallery = ({banners}) => {
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
  };
  
  return (
    <Slider {...settings}>
      {banners.map((banner: any) => {
        return (
          <div className="gallerySlide">
            <img className="gallerySlide-image"  src={banner.fields.banner} />
          </div>
        );
      })}
    </Slider>
  );
}

export default MovieGallery