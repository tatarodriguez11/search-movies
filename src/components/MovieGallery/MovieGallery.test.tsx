import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "matchmedia-polyfill";
import MovieGallery from "./MovieGallery";

describe("MovieSlider", () => {
  const movies = [
    {
      fields: {
        imageUrl: "https://example.com/image1.jpg",
      },
    },
    {
      fields: {
        banner: "https://example.com/image2.jpg",
      },
    },
  ];

  it("renders correctly with movies", () => {
    render(<MovieGallery banners={movies} />);

    // Verificar que los elementos del slider se rendericen correctamente
    expect(screen.getByRole("list")).toBeInTheDocument();
    expect(screen.getAllByRole("listitem")).toHaveLength(movies.length);

    // Verificar que los títulos y las imágenes de las películas se muestren correctamente
    movies.forEach((movie) => {
      expect(screen.queryAllByRole("img").length).toBeGreaterThan(0);
    });

    // Realizar el snapshot testing
    expect(screen.getByRole("list")).toMatchSnapshot();
  });

  it("renders correctly without movies", () => {
    render(<MovieGallery banners={[]} />);

    // Verificar que no se rendericen elementos del slider
    expect(screen.queryByRole("list")).toBeNull();
  });
});
