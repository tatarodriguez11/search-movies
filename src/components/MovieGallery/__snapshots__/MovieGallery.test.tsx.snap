// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSlider renders correctly with movies 1`] = `
<ul
  class="slick-dots"
  style="display: block;"
>
  <li
    class="slick-active"
  >
    <button>
      1
    </button>
  </li>
  <li
    class=""
  >
    <button>
      2
    </button>
  </li>
</ul>
`;
